dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/WorkflowVersionDAO.java

```diff
@@ -31,4 +34,20 @@ public WorkflowVersionDAO(SessionFactory sessionFactory) {
     public WorkflowVersion findByAlias(String alias) {
         return uniqueResult(namedQuery("io.dockstore.webservice.core.WorkflowVersion.getByAlias").setParameter("alias", alias));
     }
+
+    public List<WorkflowVersion> getWorkflowVersionsByWorkflowId(Long workflowId, int size, int firstResult) {
+        Query query = namedQuery("io.dockstore.webservice.core.WorkflowVersion.getByWorkflowId");
+        query.setParameter("id", workflowId);
+        query.setFirstResult(firstResult);
+        query.setMaxResults(size);
+        List<WorkflowVersion> workflowVersionIds = query.getResultList();
+        return workflowVersionIds;
+    }
+
+    public List<WorkflowVersion> getWorkflowVersionByWorkflowIdAndVersionName(Long workflowId, String name) {
```

Comment:
Think you can assume this is one or none due to `"unique_workflowversion_names" UNIQUE CONSTRAINT, btree (parentid, name)`

Reply:
throws exception when it's none and using singleResult, avoiding exception by using list

Reply:
Double-check classpath maybe, I have 
```
	default R getSingleResult() {
		return uniqueResult();
	}
```
which goes to 
```
	/**
	 * Convenience method to return a single instance that matches
	 * the query, or {@code null} if the query returns no results.
	 *
	 * @return the single result or <tt>null</tt>
	 *
	 * @throws NonUniqueResultException if there is more than one matching result
	 */
``` 
in hibernate-core:5.2.8-Final

Reply:
will try

Reply:
i still don't see that, the only place I see that comment is from the deprecated https://github.com/hibernate/hibernate-orm/blob/5.2.8/hibernate-core/src/main/java/org/hibernate/Query.java